### PR TITLE
Add behat scenario with free order

### PR DESCRIPTION
**Pull request by clem21**
| Q               | A
| --------------- | -----
| Branch?         | 1.7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | #11547
| License         | MIT

This is continuation of #11547

### CODE REVIEW

File: features/checkout/adding_product_with_promotion_that_makes_the_order_free.feature [hunk 1]
```diff
@@ -0,0 +1,24 @@
+@checkout
+Feature: Adding product with promotion that makes the order free
+    I want to buy a product with a promotion that gives me free shipping and free order
```
Comment (AdamKasp) [inline features/checkout/adding_product_with_promotion_that_makes_the_order_free.feature:3]: features should be described according to our convention.

File: features/checkout/adding_product_with_promotion_that_makes_the_order_free.feature [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        And there is a promotion "Holiday promotion"
+        And the promotion gives "100%" discount on shipping to every order
+        And the promotion gives "100%" discount to every order
+        And the store allows paying offline
+        And I am a logged in customer
+
+    @ui
+    Scenario: Adding a simple product to the cart
+        Given I have product "T-shirt banana" in the cart
+        And I am at the checkout addressing step
```
Comment (AdamKasp) [inline features/checkout/adding_product_with_promotion_that_makes_the_order_free.feature:18]: IMO `Given` steps should be in the `Background`

File: features/checkout/adding_product_with_promotion_that_makes_the_order_free.feature [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        When I specified the billing address as "Ankh Morpork", "Frost Alley", "90210", "United States" for "Jon Snow"
+        And I proceed with "DHL" shipping method
+        Then I should be on the checkout summary step
+        And "Holiday promotion" should be applied to my order shipping
+        Then my order total should be "$0.00"
```
Comment (AdamKasp) [inline features/checkout/adding_product_with_promotion_that_makes_the_order_free.feature:23]:
```suggestion
        And my order total should be "$0.00"
```

File: features/checkout/adding_product_with_promotion_that_makes_the_order_free.feature [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    @ui
+    Scenario: Adding a simple product to the cart
+        When I add product "T-shirt banana" to the cart
+        And I go to the checkout addressing step
+        And I specified the billing address as "Ankh Morpork", "Frost Alley", "90210", "United States" for "Jon Snow"
+        And I proceed with "DHL" shipping method
+        Then I should be on the checkout summary step
+        And "Holiday promotion" should be applied to my order shipping
+        And my order total should be "$0.00"
+        And I confirm my order
```
Comment (lchrusciel) [inline features/checkout/adding_product_with_promotion_that_makes_the_order_free.feature:26]: This is not an assertion, but rather an action. It should be part of the when step. Maybe we need scenarios for that? Or let's just make a scenario, that checks if placing a free order is possible - we should remove most of these assertions as they are not a clue.

File: features/checkout/adding_product_with_promotion_that_makes_the_order_free.feature [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        And the store has a product "T-shirt banana" priced at "$10.00"
+        And the store has "DHL" shipping method with "$50.00" fee
+        And there is a promotion "Holiday promotion"
+        And the promotion gives "100%" discount on shipping to every order
+        And the promotion gives "100%" discount to every order
+        And the store allows paying offline
+        And I am a logged in customer
+
+    @ui
+    Scenario: Adding a simple product to the cart
```
Comment (lchrusciel) [inline features/checkout/adding_product_with_promotion_that_makes_the_order_free.feature:18]: Is it a proper scenario name?
```suggestion
    Scenario: Placing free order
```
or
```suggestion
    Scenario: Buying product with promotion that makes the order free
```

File: features/checkout/adding_product_with_promotion_that_makes_the_order_free.feature [hunk 6]
```diff
@@ -0,0 +1,26 @@
+@checkout
+Feature: Adding product with promotion that makes the order free
```
Comment (lchrusciel) [inline features/checkout/adding_product_with_promotion_that_makes_the_order_free.feature:2]:
```suggestion
Feature: Buing product with promotion that makes the order free
```
file name should be adjusted as well

Comment (Zales0123) [reply features/checkout/adding_product_with_promotion_that_makes_the_order_free.feature:2]: `Buying` would be even better 💃

File: features/checkout/adding_product_with_promotion_that_makes_the_order_free.feature (hunk 4 above)
Comment (clem21) [reply features/checkout/adding_product_with_promotion_that_makes_the_order_free.feature:26]: I am not sure I understand what you are saying, should I ask another scenario that checks if placing a free order is possible and then remove the assertions in this existing file ? Or should I just change this scenario ?

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thank you, Clémence! :1st_place_medal:

**Pull request merged**